### PR TITLE
feat: move proteusService mock to datamodel support target - WPB-5719

### PR DESCRIPTION
**Pull request by netbe**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5719" title="WPB-5719" target="_blank">Sub-taskWPB-5719</a>  Make Event decoder async
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

I need to update the proteusService protocol, there are a couple of duplicated implementation.

### Causes (Optional)

We copy pasted the implementation because there was no way to share mocks

### Solutions

Move related files to WireDataModel:

- MockProteusProvider
- MockProteusServiceInterface

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by typfel — APPROVED** [review]

File: wire-ios-data-model/Tests/Source/Utils/MockProteusProvider.swift [hunk 1]
```diff
@@ -48,16 +46,21 @@ class MockProteusProvider: ProteusProviding {
...
-extension XCTestCase {
 
-    func spyForTests() -> SpyUserClientKeyStore {
-        let url = self.createTempFolder()
+    public static func spyForTests() -> SpyUserClientKeyStore {
+        let url = Self.createTempFolder()
         return SpyUserClientKeyStore(accountDirectory: url, applicationContainer: url)
     }
+
+    // FIXME: [F] this is defined in WireTesting, somehow it is not possible to import here for now
```
Comment (caldrian) [inline wire-ios-data-model/Tests/Source/Utils/MockProteusProvider.swift]: You need to link the WireTesting framework if you want to import code from it.

Screenshot 2023-12-12 at 10 37 00

File: wire-ios-data-model/Tests/Source/Utils/MockProteusProvider.swift [hunk 2]
```diff
@@ -19,25 +19,23 @@
 import Foundation
 import WireDataModel
 
-@testable import WireDataModelSupport
+public class MockProteusProvider: ProteusProviding {
```
Comment (caldrian) [inline wire-ios-data-model/Tests/Source/Utils/MockProteusProvider.swift:22]: I would prefer to move all files which belong to the same target to a unique directory tree.
For WireDataModelSupport I created a folder which is just named `Support`. Feel free to change the name.

Screenshot 2023-12-12 at 10 40 11

**Review by caldrian — APPROVED** [review]

File: wire-ios-data-model/Tests/Source/Utils/MockProteusProvider.swift (hunk 1 above)
Comment (netbe) [reply wire-ios-data-model/Tests/Source/Utils/MockProteusProvider.swift]: I tried but had other issue with XCTest then

Comment (caldrian) [reply wire-ios-data-model/Tests/Source/Utils/MockProteusProvider.swift]: There is a non-static method `createTempFolder()` defined in an extension to `XCTestCase`, btw.

Comment (netbe) [reply wire-ios-data-model/Tests/Source/Utils/MockProteusProvider.swift]: added a ticket to fix this later

**Pull request merged**